BookingSpecTests/SampleTest.feature

```diff
@@ -3,9 +3,122 @@
-	Then I wait for 5 seconds
\ No newline at end of file
+	And I set following parameters on Sign In Pop Up dialog
+		| Field    | Value      |
+		| Email    | <Email>    |
+		| Password | <Password> |
+	And I click Sign In button on PopUp
+	Then I wait while page popup is working
+	And I see message error with "<Message>"
+
+Scenarios: 
+		| Email                      | Password  | Message                                                               |
+		| testmirantistest@gmail.com | wrongpass | You entered an email address/password combination that doesn't match. |
+		| invalidEmail@g.com         | 1234qweR  | You entered an email address/password combination that doesn't match. |
+		| invalidEmail@g.com         | wrongpass | You entered an email address/password combination that doesn't match. |
+		|                            | 1234qweR  | Please enter a valid email address.                                   |
+		| testmirantistest@gmail.com |           | Please add a password                                                 |
+		|                            |           | Please enter a valid email address.                                   |
+
+	Scenario: Input valid credintials to Sign In
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	And I write email on Sign in PopUp "testmirantistest@gmail.com"
+	And I write password "1234qweR"
+	And I click Sign In button on PopUp
+	Then I see that i am Signed In
+
+Scenario: Input in email field more than 80 chars and assert that there only 80 chars
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	And I write email with length ninety chars "testmirantistesttestmirantistesttestmirantistesttestmirantistesttestmirantistest@gmail.ru"
+	Then I check that length of email is 80 chars
+
+Scenario: Attempt to recover password with invalid email
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	When I click Forgot Your Password button
+	And I write email "estmirantistest@gmail.com"
+	And I Click button Send
+	Then I wait while page popup is loading
+	Then I see an error on ForgotYourPasswordPopUp
+
+Scenario: Cancel recover password and Sign in
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	When I click Forgot Your Password button
+	And I write email "testmirantistest@gmail.com"
+	Then I click button Cancel
+	When I set following parameters on Sign In Pop Up dialog
+		| Field | Value |
+		| Email | testmirantistest@gmail.com |
+		| Password  | 1234qweR |
+	And I click Sign In button on PopUp
+	Then I see that i am Signed In
+
+Scenario: Click on recover password and Sign In with old password
+	Given I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button
+	When I click Forgot Your Password button
+	And I write email "testmirantistest@gmail.com"
+	And I Click button Send
```

Comment:
Let's use the same step 'And I click Sign In button on PopUp' but make it generic for different buttons on this dialog.